rotkehlchen/exchanges/binance.py

```diff
@@ -589,6 +607,184 @@ def _query_lending_balances(
+                        self.msg_aggregator.add_error(
+                            f'Missing key entry for {str(e)} in {self.name} {entry}. '
+                            f'Ignoring its lending interest history query.',
+                        )
+                        continue
+                    except DeserializationError as e:
+                        self.msg_aggregator.add_error(
+                            f'Error at deserializing {self.name} asset. {str(e)}. '
+                            f'Ignoring its lending interest history query.',
+                        )
+                        continue
+
+                    event = HistoryBaseEntry(
+                        event_identifier=event_identifier,
+                        sequence_index=idx,
+                        timestamp=timestamp,
+                        location=self.location,
+                        location_label=self.name,
+                        asset=asset,
+                        balance=Balance(
+                            amount=interest_received,
+                            usd_value=ZERO,
+                        ),
+                        notes=f'Interest paid from {entry["lendingType"]} {entry["productName"]} savings',  # noqa: E501
+                        event_type=HistoryEventType.RECEIVE,
+                        event_subtype=HistoryEventSubType.REWARD,
+                    )
+                    events.append(event)
+
+            if len(events) != 0:
+                with self.db.user_write() as write_cursor:
+                    try:
+                        self.history_events_db.add_history_events(write_cursor, events)
+                    except InputError as e:
+                        self.msg_aggregator.add_error(
+                            f'Failed to save Binance lending interest history {query_start_ts} '
+                            f'to {query_end_ts} in database. {str(e)}',
+                        )
+
+            with self.db.user_write() as write_cursor:
+                ranges.update_used_query_range(
+                    write_cursor=write_cursor,
+                    location_string=range_query_name,
+                    queried_ranges=[(start_ts, end_ts)] + ranges_to_query,
+                )
+
+    def query_lending_interests_history_balance(
+            self,
+    ) -> dict[str, defaultdict[AssetWithOracles, Balance]]:
+        """Queries binance lending interest history and aggregate the interest gained per asset.
+        May raise:
+        - RemoteError
+        """
+        now = ts_now()
+        balances: defaultdict[AssetWithOracles, Balance] = defaultdict(Balance)
+        interests: defaultdict[AssetWithOracles, Balance] = defaultdict(Balance)
+        filter_query = HistoryEventFilterQuery.make(
+            from_ts=BINANCE_LAUNCH_TS,
+            to_ts=now,
+            event_types=[HistoryEventType.RECEIVE],
+            event_subtypes=[HistoryEventSubType.REWARD],
```

Comment:
I would create a different subtype like interest payment because you are creating a very special subset and if for example we track in the future airdrops from binance then you will have an issue querying them. What do you think?